src/fides/api/service/connectors/rds_mysql_connector.py

```diff
@@ -0,0 +1,153 @@
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_rds_mysql import (
+    RDSMySQLSchema,
+)
+from fides.api.service.connectors.query_config import MySQLQueryConfig, SQLQueryConfig
+from fides.api.service.connectors.rds_connector_mixin import RDSConnectorMixin
+from fides.api.service.connectors.sql_connector import SQLConnector
+from fides.api.util.collection_util import Row
+
+CA_CERT_URL = "https://truststore.pki.rds.amazonaws.com/global/global-bundle.pem"
+
+
+class RDSMySQLConnector(RDSConnectorMixin, SQLConnector):
+    """
+    Connector specific to RDS MySQL databases
+    """
+
+    secrets_schema = RDSMySQLSchema
+    namespace_meta: Optional[dict] = None
+
+    def build_uri(self) -> Optional[str]:
+        """
+        We need to override this method so it is not abstract anymore, and RDSMySQLConnector is instantiable.
+        """
+
+    @cached_property
+    def typed_secrets(self) -> RDSMySQLSchema:
+        return self.secrets_schema(**self.configuration.secrets or {})
+
+    @property
+    def url_scheme(self) -> str:
+        """
+        Returns the URL scheme for the monitor's Engine.
+        """
+        return "mysql+pymysql"
+
+    @property
+    def aws_engines(self) -> list[str]:
+        """
+        Returns the AWS engines supported by the monitor.
+        """
+        return ["mysql", "aurora-mysql"]
+
+    def pre_client_creation_hook(self, node: ExecutionNode) -> None:
+        """
+        Pre client hook for RDS MySQL Connector
+        """
+        db: Session = Session.object_session(self.configuration)
+        self.namespace_meta = SQLConnector.get_namespace_meta(db, node.address.dataset)
+
+    def get_db_name_and_schema_name(self) -> tuple[str, str]:
+        """
+        Returns the database name and schema name for the provided staged resource.
+        """
+        if self.namespace_meta is None:
+            raise ConnectionException(
+                "Namespace meta is not set. Please call pre_client_creation_hook before creating the client."
+            )
+        return (
+            self.namespace_meta["database_instance_id"],
+            self.namespace_meta["database_id"],
+        )
+
+    # Overrides SQLConnector.create_client
+    def create_client(self) -> Engine:
+        """
+        Returns a SQLAlchemy Engine that can be used to interact with a database
+        """
+        database_instance_name, db_name = self.get_db_name_and_schema_name()
+        db_info = self.get_database_instance_connection_info(database_instance_name)
+        return self.create_engine(
+            db_username=self.typed_secrets.db_username,
+            host=db_info["host"],
+            port=db_info["port"],
+            db_name=db_name,
+        )
```

Comment:
I know we're not using this code for the DSR anymore but I think it's still worth calling out that it seems a bit fragile to me. I think we should try to put as much logic as we can inside of `create_client` and `get_client`. We could potentially do something where we set the current node on the `RDSMySQLConnector` instance and have `create_client` and `get_client` have access to that. We could also calculate the `db_info` during `__init__()` so that every time we need to create a client we can just look up the host and port.